lib/applicationProcessing/resolvers.ts

```diff
@@ -0,0 +1,237 @@
+    throw new ApolloError('Application Processing object was unable to be updated');
+  }
+
+  return {
+    ok: true,
+  };
+};
+
+/**
+ * Completes an application by setting the ApplicationStatus to COMPLETED, querying the application data,
+ * and calling updateApplicant, updateGuardian, updateMedicalInformation, and upsertPhysician with the queried data.
+ * @param {ID!} args.applicationId - The id of the Application to complete
+ * @param {ID!} args.applicationProcessingId - The id of the ApplicationProcessing object
+ * @returns Status of operation (ok, error)
+ */
+export const completeApplication: Resolver = async (_, args, { prisma }) => {
+  const { applicationId } = args;
+
+  try {
+    const application = await prisma.application.findUnique({
+      where: { id: parseInt(applicationId) },
+      include: { applicationProcessing: true },
+    });
+
+    if (!application) {
+      throw 'Application not found.';
+    }
+
+    const {
+      isRenewal,
+
+      rcdUserId,
+      firstName,
+      lastName,
+      gender,
+      customGender,
+      dateOfBirth,
+      email,
+      phone,
+      province,
+      city,
+      addressLine1,
+      addressLine2,
+      postalCode,
+
+      disability,
+      affectsMobility,
+      mobilityAidRequired,
+      cannotWalk100m,
+
+      guardianFirstName,
+      guardianMiddleName,
+      guardianLastName,
+      guardianPhone,
+      guardianProvince,
+      guardianCity,
+      guardianAddressLine1,
+      guardianAddressLine2,
+      guardianPostalCode,
+      guardianRelationship,
+      guardianNotes,
+
+      physicianName,
+      physicianMspNumber,
+      physicianAddressLine1,
+      physicianAddressLine2,
+      physicianCity,
+      physicianProvince,
+      physicianPostalCode,
+      physicianPhone,
+      physicianNotes,
+
+      applicantId,
+      applicationProcessing,
+    } = application;
+
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    const applicantData = {
+      rcdUserId,
+      firstName,
+      lastName,
+      dateOfBirth,
+      gender,
+      customGender,
+      email,
+      phone,
+      province,
+      city,
+      addressLine1,
+      addressLine2,
+      postalCode,
+    };
+
+    const medicalInformationData = {
+      disability,
+      affectsMobility,
+      mobilityAidRequired,
+      cannotWalk100m,
+    };
+
+    const physicianData = {
+      name: physicianName,
+      mspNumber: physicianMspNumber,
+      addressLine1: physicianAddressLine1,
+      addressLine2: physicianAddressLine2,
+      city: physicianCity,
+      province: physicianProvince,
+      postalCode: physicianPostalCode,
+      phone: physicianPhone,
+      notes: physicianNotes,
+    };
+
+    const promises = [];
+    const applicationProcessingResult = prisma.applicationProcessing.update({
+      where: { id: applicationProcessing?.id },
+      data: {
+        status: 'COMPLETED',
+      },
+    });
+    promises.push(applicationProcessingResult);
```

Comment:
Move this to after the `if` condition below. Easier to read/understand because status update should only occur if the applicant was successfully updated.